java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogSnapshotStore.java

```diff
@@ -0,0 +1,212 @@
+                        "#UpdateTime = :update_time")
+                .withConditionExpression("#TransactionNumber <> :transaction_number")
+                .withExpressionAttributeNames(Map.of(
+                        "#Path", pathKey,
+                        "#TransactionNumber", transactionNumberKey,
+                        "#UpdateTime", UPDATE_TIME))
+                .withExpressionAttributeValues(Map.of(
+                        ":path", createStringAttribute(snapshotPath),
+                        ":transaction_number", createNumberAttribute(transactionNumber),
+                        ":update_time", createNumberAttribute(updateTime.toEpochMilli())));
+    }
+
+    public List<TransactionLogSnapshot> getFilesSnapshots() {
+        return getSnapshots(SnapshotType.FILES)
+                .collect(Collectors.toList());
+    }
+
+    public List<TransactionLogSnapshot> getPartitionsSnapshots() {
+        return getSnapshots(SnapshotType.PARTITIONS)
+                .collect(Collectors.toList());
+    }
+
+    private Stream<TransactionLogSnapshot> getSnapshots(SnapshotType type) {
+        return streamPagedItems(dynamo, new QueryRequest()
+                .withTableName(allSnapshotsTable)
+                .withConsistentRead(true)
+                .withKeyConditionExpression("#TableIdAndType = :table_and_type")
+                .withExpressionAttributeNames(Map.of("#TableIdAndType", TABLE_ID_AND_SNAPSHOT_TYPE))
+                .withExpressionAttributeValues(new DynamoDBRecordBuilder()
+                        .string(":table_and_type", tableAndType(allSnapshotsTable, type))
+                        .build())
+                .withScanIndexForward(false)
+                .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL))
+                .map(DynamoDBTransactionLogSnapshotStore::getSnapshotFromItem);
+    }
+
+    public Optional<TransactionLogSnapshot> getLatestFilesSnapshot() {
+        return getLatestSnapshot(SnapshotType.FILES);
+    }
+
+    public Optional<TransactionLogSnapshot> getLatestPartitionsSnapshot() {
+        return getLatestSnapshot(SnapshotType.PARTITIONS);
+    }
+
+    private Optional<TransactionLogSnapshot> getLatestSnapshot(SnapshotType snapshotType) {
+        return streamPagedItems(dynamo, new QueryRequest()
+                .withTableName(latestSnapshotsTable)
+                .withConsistentRead(true)
+                .withKeyConditionExpression("#TableId = :table_id")
+                .withExpressionAttributeNames(Map.of("#TableId", TABLE_ID))
+                .withExpressionAttributeValues(new DynamoDBRecordBuilder()
+                        .string(":table_id", sleeperTableId)
+                        .build())
+                .withScanIndexForward(false)
+                .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL))
+                .map(item -> getLatestSnapshotFromItem(item, snapshotType))
+                .findFirst();
```

Comment:
The streamPagedItems and findFirst seem wrong here, since this should only be one item.